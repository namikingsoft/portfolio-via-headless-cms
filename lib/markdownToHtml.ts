import { remark } from 'remark'
import html from 'remark-html'
import prism from 'remark-prism'
import gfm from 'remark-gfm'
import toc from 'remark-toc'
import math from 'remark-math'
import emoji from 'remark-emoji'
import simplePlantUML from '@akebifiky/remark-simple-plantuml'

export default async function markdownToHtml(markdown: string) {
  const result = await remark()
    .use(gfm)
    .use(toc, { heading: 'toc|TOC|Table of Contents|目次', maxDepth: 2 })
    .use(math)
    .use(simplePlantUML)
    .use(emoji)
    .use(prism)
    .use(html, { sanitize: false }) // cms content only
    .process(markdown)
  return result.toString()
}
