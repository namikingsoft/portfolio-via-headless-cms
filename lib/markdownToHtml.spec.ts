import markdownToHtml from './markdownToHtml'

test('image with dimension', async () => {
  const html = await markdownToHtml(`![img {640x480}](//example.com/test.png)`)
  expect(html).toMatchSnapshot()
})

test('image without dimension', async () => {
  const html = await markdownToHtml(`![img](//example.com/test.png)`)
  expect(html).toMatchSnapshot()
})
