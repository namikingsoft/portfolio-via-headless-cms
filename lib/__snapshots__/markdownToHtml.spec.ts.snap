// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image with dimension 1`] = `"<p><img alt="img {640x480}" srcSet="/_next/image?url=https%3A%2F%2Fexample.com%2Ftest.png&amp;w=640&amp;q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Ftest.png&amp;w=1920&amp;q=75 2x" src="/_next/image?url=https%3A%2F%2Fexample.com%2Ftest.png&amp;w=1920&amp;q=75" width="640" height="480" decoding="async" data-nimg="1" loading="lazy" style="color:transparent"/></p>"`;

exports[`image without dimension 1`] = `"<p><img src="//example.com/test.png" alt="img"/></p>"`;
